### PR TITLE
fix: fix issue in graph for handling charge state

### PR DESCRIPTION
**Pull request by CodeBear801**
issue: https://github.com/Telenav/osrm-backend/issues/143

## Tasklist

 - [x] add tests 
 - [x] review
 - [x] adjust for comments

### CODE REVIEW

File: integration/oasis/stationgraph/graph_test.go [hunk 1]
```diff
@@ -644,191 +644,191 @@ func TestGraphGeneral(t *testing.T) {
...
-							chargeEnergy:  0,
-						},
-					},
-				},
-				startNodeID: 0,
-				endNodeID:   8,
-			},
-			[]nodeID{1, 4, 5},
-		},
+		// {
```
Comment (CodeBear801) [inline integration/oasis/stationgraph/graph_test.go]: This test will be re-enabled when merge with other pull, for full context please go to: https://github.com/Telenav/osrm-backend/pull/203

File: integration/oasis/stationgraph/node.go [hunk 2]
```diff
@@ -33,13 +40,29 @@ func newNode() *node {
 	}
 }
 
+// Function isLocationReachable is used to test whether target node is reachable
```
[the remark this answers is not in the document]
Comment (CodeBear801) [reply integration/oasis/stationgraph/node.go]: https://github.com/Telenav/osrm-backend/issues/208

File: integration/oasis/stationgraph/node.go [hunk 3]
```diff
@@ -33,13 +40,29 @@ func newNode() *node {
...
+}
+
+// calcChargeTime calculates time effort needed from previous final status to current
+func (n *node) calcChargeTime(prev *node, distance float64, strategy chargingstrategy.ChargingStrategyCreator) float64 {
+	arrivalEnergy := prev.chargeEnergy - distance
+	if arrivalEnergy < 0 {
+		glog.Fatalf("Before updateNode should check isLocationReachable() prev.arrivalEnergy=%#v distance=%#v", prev.arrivalEnergy, distance)
+	}
+	//return strategy.EvaluateCost(arrivalEnergy, chargingstrategy.ChargingStatus{ChargingEnergy: n.chargeEnergy}).Duration
+	return 0.0
```
Comment (wangyoucao577) [inline integration/oasis/stationgraph/node.go]: want to return `0.0`?

Comment (CodeBear801) [reply integration/oasis/stationgraph/node.go]: Want to return 
```
//return strategy.EvaluateCost(arrivalEnergy, chargingstrategy.ChargingStatus{ChargingEnergy: n.chargeEnergy}).Duration
```
But need other pull to merge, to keep this atomic pull request work, return `0.0`.

Comment (wangyoucao577) [reply integration/oasis/stationgraph/node.go]: Fair enough.

**Pull request merged**